qiskit/algorithms/evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -30,29 +32,63 @@ class VarQTELinearSolver:
+        gradient_params: List[Parameter],
         evolution_grad: OperatorBase,
+        t_param=None,
         lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] = np.linalg.lstsq,
-        circuit_sampler: Optional[CircuitSampler] = None,
+        quantum_instance: Optional[QuantumInstance] = None,
         imag_part_tol: float = 1e-7,
     ) -> None:
         """
         Args:
-            metric_tensor: A parametrized operator that represents the left-hand side of an ODE.
+            ansatz: Quantum state in the form of a parametrized quantum circuit.
+            qfi: A Quantum Fisher Information instance used to calculate a metric tensor for the
+                left-hand side of an ODE.
+            gradient_params: List of parameters with respect to which gradients should be computed.
             evolution_grad: A parametrized operator that represents the right-hand side of an ODE.
+            t_param: Time parameter in case of a time-dependent Hamiltonian.
             lse_solver: Linear system of equations solver that follows a NumPy
                 ``np.linalg.lstsq`` interface.
-            circuit_sampler: Samples circuits using an underlying backend.
+            quantum_instance: Backend used to evaluate the quantum circuit outputs. If ``None``
+                provided, everything will be evaluated based on matrix multiplication (which is
+                slow).
             imag_part_tol: Allowed value of an imaginary part that can be neglected if no
                 imaginary part is expected.
         """
-
-        self._metric_tensor = metric_tensor
+        self._ansatz = ansatz
+        self._qfi = qfi
+        self._gradient_params = gradient_params
+        bind_params = gradient_params + [t_param] if t_param else gradient_params
+        self._qfi_gradient_callable = qfi.gradient_wrapper(
+            CircuitStateFn(ansatz), bind_params, gradient_params, quantum_instance
+        )
         self._evolution_grad = evolution_grad
+        self._time_param = t_param
         self._lse_solver = lse_solver
-        self._circuit_sampler = circuit_sampler
+        self._quantum_instance = None
+        self._circuit_sampler = None
+        if quantum_instance is not None:
+            self.quantum_instance = quantum_instance
         self._imag_part_tol = imag_part_tol
 
+    @property
+    def quantum_instance(self) -> Optional[QuantumInstance]:
+        """Returns quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Union[QuantumInstance, Backend]) -> None:
+        """Sets quantum_instance"""
+        if not isinstance(quantum_instance, QuantumInstance):
+            quantum_instance = QuantumInstance(quantum_instance)
+
+        self._quantum_instance = quantum_instance
+        self._circuit_sampler = CircuitSampler(
+            quantum_instance, param_qobj=is_aer_provider(quantum_instance.backend)
+        )
```

Comment:
If we need to change the quantum instance we should also update the QFI callable -- but do we need to change it at all? None of the other attributes have setters/getters 🙂

Reply:
You are right.
Well, this is not even directly used by a user so the quantum instance will likely never be changed. I just added this option because Steve was usually advocating for having it to be compliant with Qiskit conventions...

Reply:
Thanks for clarifying. In this case I will remove it.